### ISSUE TITLE
the hint for joining order function was invalid on queries with more than 11 tables

### ISSUE THREAD

**Issue by ohlala** [ossc-db/pg_hint_plan#36]
Hello,

When I experimented with the join order benchmark using pg_hint_plan, I found that the hint for joining order function was invalid on queries with more than 11 tables.

These queries include (29c 29b 29a 33c 33b 33a 28c 28b 28a 30c 30b 30a 27c 27b 27a 26c 26b 26a 24b 24a ).

For instance, when I used the following hint command for 33a, there was no change in the query plan.
```
 /*+  Leading(  ((((((kt ((it (((it1 ((lt ml) mi_idx1)) mc1) cn1)) t)) mc) cn) t1) kt1) mi_idx) )  */
 explain ...

/*+  Leading(  ((((((kt ((it (((it1 ((ml lt) mi_idx1)) mc1) cn1)) t)) mc) cn) t1) kt1) mi_idx) )  */
 explain ...

/*+  Leading(  (((((( (( (((it1 ((ml lt) mi_idx1)it) mc1) cn1)kt) t)) mc) cn) t1) kt1) mi_idx) )  */
 explain ...

```
But for the query of <=11 tables, the result is in line with expectations.

Is this feature limited to the number of tables? Or is there any problem with the method I am using?

My experimental environment is as follows:
PostgreSQL 10.7
Ubuntu 16.04
pg_hint_plan--1.3.3 Download from this link (https://ja.osdn.net/projects/pghintplan/scm/git/pg_hint_plan/tree/PG10/)

I wonder if you can give me some advice. Thank you~!

**Comment by yamatattsu** [comment]
Hi KUN,

>Is this feature limited to the number of tables? Or is there any problem
with the method I am using?
pg_hint_plan doesn't handle Join order and join method hints, if you use
default value of these parameters:
  - from_collapse_limit
  - join_collapse_limit
  - geqo_threshold
Therefore, you would be better to increase these numbers at least 11 or 12.
If my memory is correct, I used these number: 15 or 16 when I measured a
performance of pg_plan_advsr using Join order benchmark.

Thanks,
Tatsuro Yamada



2019年6月12日(水) 17:26 KUN <notifications@github.com>:
